Spatial_Engine/Query/IsSelfIntersecting.cs

```diff
@@ -34,6 +37,10 @@ public static partial class Query
         /****            IElement1D            ****/
         /******************************************/
 
+        [Description("Returns if the one dimensional representation of the IElement1D is closer to itself than the tolerance at any two points.")]
+        [Input("element1D", "The IElement1D to evaluate self intersections from.")]
+        [Input("tolerance", "Minimum distance to be considered intersecting.", typeof(Length))]
+        [Output("o", "A boolean which is true if the IElement1Ds curve is self intersecting.")]
```

Comment:
not sure I get the "o"?

Reply:
output, seemed silly to write `isSelfIntersecting` again

Reply:
I think for consistency of approach let's keep the full output name `isSelfIntersecting`.

Actually agree on the `OutputAttribute` there is duplication of information - forced from the format of the constructor requiring `name` & `description`  
``` C#
OutputAttribute(string name, string description, Type quantity = null)`
```

One to review more generally, but for now for clarity let us simply stick with full output name.
@IsakNaslundBh @kThorsager sound OK?

Reply:
Changed it and am OK, but should perhaps waited for happiness confirmation from @IsakNaslundBh first

Reply:
Sorry - 🙈 just realised we have the constructor _already_ for OutputAttribute that takes a single string for description only. 👍 

``` C#

public OutputAttribute(string description)

```

@kThorsager  suggest you use this instead!

Reply:
oh, when is the general case for not specifying a name?